__tests__/app.test.js

```diff
@@ -167,6 +159,64 @@ describe("GET /api/articles/:article_id/comments", () => {
   });
 });
 
+describe("POST /api/articles/:article_id/comments", () => {
+  test("201: Returns 201 when post has been added to the article of give article_id ", () => {
+    
+    const input = { username: "rogersop", body: "In my humble opinion, I prefer beans." }
+    
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(input)
+      .expect(201)
+      .then(({ body }) => {
+          expect(body.comment[0]).toEqual({
+            comment_id: expect.any(Number),
+            article_id: 2,
+            author: "rogersop",
+            body: "In my humble opinion, I prefer beans.",
+            votes: 0,
+            created_at: expect.any(String),
+          });
+      });
+    });
+  test("400: Returns 400 error if no body provided", () => {
+    
+    const input = { username: "rogersop" }
+    
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(input)
+      .expect(400)
+      .then(({ body }) => {
+          expect(body.msg).toBe("Bad request - must have a valid username and a body");
+        });
+   });
+   test("400: Returns 400 error if no username provided", () => {
+    
+    const input = { body: "In my humble opinion, I prefer beans." }
+    
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(input)
+      .expect(400)
+      .then(({ body }) => {
+          expect(body.msg).toBe("Bad request - must have a valid username and a body");
+        });
+   });
+   test("400: Returns 400 error if article doesn't exist or is invalid", () => {
+    
+    const input = { username: "rogersop", body: "In my humble opinion, I prefer beans." }
+    
+    return request(app)
+      .post("/api/articles/a1/comments")
+      .send(input)
+      .expect(400)
+      .then(({ body }) => {
+          expect(body.msg).toBe("Bad request - invalid input");
+        });
+   });
+});
```

Comment:
This is a good start to error handling, but there are a couple more important error handling tests to be added:

When given an article id that is valid but does not exist:
`/api/articles/9999/comments`

When given a username that is valid but does not exist:
`{ username: "bob", body: "In my humble opinion, I prefer bananas." }`